src/Button/Button.js

```diff
@@ -0,0 +1,102 @@
+import React, {Component, PropTypes} from 'react';
+import propTypes from './buttonPropTypes';
+import EnhancedButton from '../internal/EnhancedButton';
+import getStyles from './buttonStyles';
+import styleButtonChildren from './styleButtonChildren';
+
+import {
+  handleMouseDown,
+  handleMouseUp,
+  handleMouseLeave,
+  handleMouseEnter,
+  handleTouchStart,
+  handleTouchEnd,
+  handleKeyboardFocus,
+} from './buttonEventHandlers';
+
+class Button extends Component {
+  static muiName = 'Button';
+  static propTypes = propTypes;
+
+  static defaultProps = {
+    disabled: false,
+    fullWidth: false,
+    onKeyboardFocus: () => {},
+    onMouseDown: () => {},
+    onMouseEnter: () => {},
+    onMouseLeave: () => {},
+    onMouseUp: () => {},
+    onTouchEnd: () => {},
+    onTouchStart: () => {},
+    primary: false,
+    secondary: false,
+    type: 'raised',
```

Comment:
Reading the spec, I have the feeling that the type of the button is something that matters quite a lot.
What about making this property mandatory with no default? That would force users to explicit the type of button they want.